presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -37,6 +39,8 @@
     private boolean cacheMissing;
 
     @NotNull
+    // Some drivers match case insensitive in Driver.acceptURL
+    @Pattern(regexp = "^jdbc:[a-z0-9]+:(?s:.*)$", flags = CASE_INSENSITIVE, message = "Invalid JDBC URL: has not jdbc:<subprotocol>:<subname> format.")
```

Comment:
I didn't know about the case insensitivity.  Also, it feels like not needed, since it matters for jdbc prefix only, but whatever.

What about using a shorter pattern, like:

```
jdbc:(?:.*)
```

i intentionally omitted the subprotocol part
also, the `^` and `$` are not required, as pattern is _matched_

then you can also omit the message, as the auto-generated one should be usable (is it?)

Reply:
This will accept url `jdbc:` which is invalid from the spec perspective.

Reply:
I've adjusted message to include regexp and wording from original `Pattern` message

Reply:
nice, i didnt know you can do `{regex}`
And what would happen if you do not specify `message` at all?
I remember i didn't have to do into some other project, but maybe there was something extra done there

Reply:
The default message will be used which is `{javax.validation.constraints.Pattern.message}` which interpolates to `must match the following regular expression: {regexp}`